text/0000-drop-types-in-const.md

```diff
@@ -0,0 +1,68 @@
+- Feature Name: `drop_types_in_const`
+- Start Date: 2016-01-01
+- RFC PR: (leave this empty)
+- Rust Issue: (leave this empty)
+
+# Summary
+[summary]: #summary
+
+Allow types with destructors to be used in `static` items and in `const` functions, as long as the destructor never needs to run in const context.
+
+# Motivation
+[motivation]: #motivation
+
+Most collection types do not allocate any memory when constructed empty. With the change to make leaking safe, the restriction on `static` items with destructors
+is no longer required to be a hard error.
+
+Allowing types with destructors to be directly used in `const` functions and stored in `static`s will remove the need to have
+runtime-initialisation for global variables.
+
+# Detailed design
```

Comment:
This section should explore different implementation strategies as well. E.g. do we use `.ctors` and `.dtors` sections (that’s what c++ does) which allows for life-before-main (which we’ve been very reluctant about allowing) or do we employ scheme similar to `lazy_static!` (initialize on first use)?